### PR TITLE
Add info about the gap limit to the glossary

### PR DESCRIPTION
**Pull request by GBKS**
Addresses #357.

[Preview](https://deploy-preview-764--sad-borg-390916.netlify.app/guide/glossary/#gap-limit)

### CODE REVIEW

File: guide/glossary/index.md [hunk 1]
```diff
@@ -188,6 +188,10 @@ Same as xpub, however the y denotes that this xpub belongs to a wallet that is f
 
 Same as ypub though the z denotes it is an extended public key from a segregated witness enabled wallet following [BIP84](https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki).
 
+### Gap limit
+
+Wallets generally create 20 addresses and watch them for incoming transactions. As addresses are used, new ones are generated. If a wallet is imported into an application and it has received bitcoin on addresses 21 and higher, then those will not be found initially. This may cause problems for users. For more, see [BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Address_gap_limit).
```
Comment (pavlenex) [inline guide/glossary/index.md:193]: Wallets generally create 20 addresses and watch them for incoming transactions. As addresses are used, new ones are generated. If a wallet is imported into an application and it has received bitcoin on addresses 21 and higher, then those will not be found initially

A few problems here:
- Needs to be made clear this is for on-chain wallets.
- We should add reason why wallets do this (there's no good reason except performance, meaning a lot of users query the same node for balance, so they limit it.
- As addresses are **used**, new ones are generated. - I think `used` here gives wrong impression. The gap limit happens if you have 20 unused addresses consequently generated and 21st receives payment. So 20 unused empty addresses.

I tried to explain this issue more practically here https://docs.btcpayserver.org/FAQ/Wallet/#the-gap-limit-solution

**Comment by Bosch-0** [comment]
For a little less technical read to reference here: https://blog.lopp.net/mind-the-bitcoin-address-gap/

File: guide/glossary/index.md (hunk 1 above)
Comment (GBKS) [reply guide/glossary/index.md]: Thank you. Just implemented your feedback.

**Comment by GBKS** [comment]
Thanks for the feedback. Just pushed an update. Let me know how it looks now.

**Review by Bosch-0 — APPROVED** [review]
LGTM

**Review by sbddesign — APPROVED** [review]
LGTM

**Pull request merged**